src/azure-cli/azure/cli/command_modules/synapse/_help.py

```diff
@@ -901,15 +923,24 @@
   - name: Create a role assignment using service principal name.
     text: |-
         az synapse role assignment create --workspace-name testsynapseworkspace \\
-        --role "Sql Admin" --assignee sp_name
```

Comment:
does `Sql Admin` still work?

Reply:
No, now Synapse is using new role name. 
Synapse SQL Administrator
Synapse Apache Spark Administrator
...

https://docs.microsoft.com/en-us/azure/synapse-analytics/security/synapse-workspace-synapse-rbac-roles

Reply:
In this way, it is breaking change to existing customers and we should try to avoid such breaking. 
To not break customers, you should try to make previous definition work and add deprecation info to redirect to new definition.

Reply:
Can we just allow this breaking change considering that the module is still in preview? Now Synapse moves to new names, we don't want to allow the old names.

Reply:
Adding breaking change description at History Notes

[Synapse] BREAKING CHANGE: az synapse role assignment create: Role names at old version are not allowed, Sql Admin, Apache Spark Admin, Workspace Admin

Reply:
I see. According to the telemetry, there is no much usage for this command as it is in preview. So we could accept the breaking change this time. Please note that we need to avoid breaking change as possible as we can in future design.